packages/sdk/src/libs/stores/index.ts

```diff
@@ -11,4 +12,9 @@ export interface Store<I, D> {
   values: () => Promise<Array<D>>;
   keys: () => Promise<Array<I>>;
   size: () => Promise<number>;
+  clear: () => Promise<void>;
+  // these fields are for stores with sorted data
+  between: (a: I, b: I) => Promise<Array<D>>;
+  slice: (id: I, length: number) => Promise<Array<D>>;
```

Comment:
A few comments about these new functions:

- Does this mean that Ids need to have some sort of defined ordering? Should/can we augment the type to enforce this?
- If we have instances of this interface that _don't_ implement some of these (or error when you call them), should they be part of the same interface? It seems like this breaks the abstraction and requires the caller to be aware of what underlying store type they're using, right?

I could see us keeping the abstraction by defining different interfaces, so we can have users specify exactly what interface they need at compile time:
```ts
export interface BaseStore<I, D> {
  get: (id: I) => Promise<D | undefined>;
  entries: () => Promise<Array<[I, D]>>;
  values: () => Promise<Array<D>>;
  size: () => Promise<number>;
}

export interface KVStore<I, D> extends BaseStore<I, D> {
  ...
}

export interface OrderedStore<I, D> extends BaseStore<I, D> {
  ...
}
```

Or if you think every Ordered store can do everything a KV store can do (which isn't obviously true for things like random deletes), you could make OrderedStore extend KVStore.

Thoughts?

Reply:
> Does this mean that Ids need to have some sort of defined ordering? Should/can we augment the type to enforce this?

as long as ids are strings or numbers, they can be sorted just fine. arrays or objects would not work, maybe this needs to be clarified.

> If we have instances of this interface that don't implement some of these (or error when you call them), should they be part of the same interface? It seems like this breaks the abstraction and requires the caller to be aware of what underlying store type they're using, right?

Yea you are right, i kind of mashed things into the same interface.  originally tried what you are suggesting, breaking up the interfaces, but thought it didn't gain much because i only needed the 2 additional functions.  but what you are suggesting i think is the right way to do this. I think i will refactor it based on this suggestion.